### ISSUE TITLE
Pets love to do it from behind

### ISSUE THREAD

**Issue by ipriver** [TrinityCore/TrinityCore#24894]
**Description:**
Not sure what is causing it, but as you can see in my recorded video [here](https://imgur.com/a/656uG2v) pet always tries to stab you only from behind.

Tested with a wolf and a spider pet only in duels.

**Expected behaviour:**

Pet doesn't try to be behind and attacks from the nearest position.

**Steps to reproduce the problem:**

1. Start a duel with a class who has a pet
2. Command your pet to attack

**Branch(es):**

3.3.5

**TC rev. hash/commit:** 

3ac1992afbeb

**Operating system:** 

Ubuntu 20.04

**Comment by Aokromes** [comment]
not a bug.
https://github.com/TrinityCore/TrinityCore/issues/19925

**Comment by ipriver** [comment]
@Aokromes 
https://youtu.be/G_m9hwU75No?t=268 in this old WotLK video can be seen that pet doesn't always try to be behind targets at least during pvp fights.

**Comment by Killyana** [comment]
Yes this looks wrong, maybe only the initial positioning must be behind the target then no update must be done.

**Comment by jackpoz** [comment]
Well we have 2 different issues that say different things do better figure out how it's supposed to be first.

**Comment by Ovahlord** [comment]
That's just a result of blizzard updating chase movement every 400ms while we have hardcoded  this behaivior to attack from behind via chase angle. Just have to drop that

**Comment by Jildor** [comment]
We can see in that video how pet is tanking and does not seek the back of the target:
https://www.youtube.com/watch?v=T5jVVCJZw2Q

and here a pvp video:
https://www.youtube.com/watch?v=_BWeGKTnUUg

in my opinion, I think pet should seek the back of the target if chaseDistance is 0 (pet without ranged attacks) and isn't a player and pettarget hasn't as a victim the pet.
or something like that

**Comment by ipriver** [comment]
> #19921 

The point of this ticket was to give an advantage of pets to attack from behind so that mobs won't parry their attacks (dodge works anyway). OK, but:
- in PvP this fix is overpowered as players can't dodge/parry from behind and they constantly are getting damage (hello rogues).
- in PvE some dragon bosses have tail damage and pushback for those who are attacking from the back (e.g. Onyxia) which makes pets to be useless.

**Comment by natnat201193** [comment]
Pet vs NPC (melee pets):
Pet is Not Victim? Attack from the side (any side) if a path to said angle is available, if not, normal chase movement(Kologarn is a good example where the side path is not available).
Pet is Victim? Normal chase movement (no angle seeking).

Pet vs player:
Normal Chase movement (no angle seeking).

This is how it was in wrath and this is how it still is to this day on retail.

@Ovahlord > blizzard updating chase movement every 400ms
This delay can actually be seen on retail still, if you build some threat on a mob and have it chase you around while your pet is attacking, you can sometimes see a slight delay before the pet will adjust the angle.
